### ISSUE TITLE
Improve error messages with no/invalid API key

### ISSUE THREAD

**Issue by StevenShockley** [nextcloud/weather#38]
The error message provided when the API key is invalid doesn't point to the issue.

### Steps to reproduce
1. Open weather app.  Delete any existing cities, and set API key to 0 (if already set).  Switch to another app and back if needed to clear the page.
2. Search for a city.  App returns "Fatal error: please check your owncloud.log and send a bug report here: https://github.com/nextcloud/weather/issues"
3. These logs appear in nextcloud.log:
```
{"reqId":"VrwbgcOxaIB9z30QD483","level":3,"time":"2017-06-24T07:15:03+00:00","remoteAddr":"","user":"--","app":"PHP","method":"--","url":"--","message":"Cannot retrieve value of undefined directive HTML.TargetNoreferrer invoked on line 276 in file \/var\/www\/nextcloud\/apps\/news\/vendor\/ezyang\/htmlpurifier\/library\/HTMLPurifier\/HTMLModuleManager.php at \/var\/www\/nextcloud\/apps\/news\/vendor\/ezyang\/htmlpurifier\/library\/HTMLPurifier\/Config.php#901","userAgent":"--","version":"12.0.0.29"}
{"reqId":"VrwbgcOxaIB9z30QD483","level":3,"time":"2017-06-24T07:15:03+00:00","remoteAddr":"","user":"--","app":"PHP","method":"--","url":"--","message":"Cannot retrieve value of undefined directive HTML.TargetNoopener invoked on line 279 in file \/var\/www\/nextcloud\/apps\/news\/vendor\/ezyang\/htmlpurifier\/library\/HTMLPurifier\/HTMLModuleManager.php at \/var\/www\/nextcloud\/apps\/news\/vendor\/ezyang\/htmlpurifier\/library\/HTMLPurifier\/Config.php#901","userAgent":"--","version":"12.0.0.29"}
{"reqId":"VrwbgcOxaIB9z30QD483","level":3,"time":"2017-06-24T07:15:03+00:00","remoteAddr":"","user":"--","app":"PHP","method":"--","url":"--","message":"Cannot retrieve value of undefined directive Core.LegacyEntityDecoder invoked on line 337 in file \/var\/www\/nextcloud\/apps\/news\/vendor\/ezyang\/htmlpurifier\/library\/HTMLPurifier\/Lexer.php at \/var\/www\/nextcloud\/apps\/news\/vendor\/ezyang\/htmlpurifier\/library\/HTMLPurifier\/Config.php#901","userAgent":"--","version":"12.0.0.29"}
{"reqId":"VrwbgcOxaIB9z30QD483","level":3,"time":"2017-06-24T07:15:03+00:00","remoteAddr":"","user":"--","app":"PHP","method":"--","url":"--","message":"Cannot retrieve value of undefined directive Core.AggressivelyRemoveScript invoked on line 352 in file \/var\/www\/nextcloud\/apps\/news\/vendor\/ezyang\/htmlpurifier\/library\/HTMLPurifier\/Lexer.php at \/var\/www\/nextcloud\/apps\/news\/vendor\/ezyang\/htmlpurifier\/library\/HTMLPurifier\/Config.php#901","userAgent":"--","version":"12.0.0.29"}
```
The app works fine once I enter the API key.

### Expected behaviour
Error should mention the API key

### Actual behaviour
Error in logs that points to the news app and htmlpurifier, which was confusing at first

### Server configuration
techandme.se Nextcloud appliance

**Nextcloud version:** (see Nextcloud admin page)
12.0.0

**Updated from an older Nextcloud/ownCloud or fresh install:**
Updated from Nextcloud 11

**Where did you install Nextcloud from:**
techandme.se appliance

**Signing status:**
<details>
<summary>Signing status</summary>

```
No errors have been found.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>
Enabled:
  - activity: 2.5.2
  - admin_audit: 1.2.0
  - bruteforcesettings: 1.0.2
  - calendar: 1.5.3
  - comments: 1.2.0
  - contacts: 1.5.3
  - dav: 1.3.0
  - deck: 0.2.0
  - federatedfilesharing: 1.2.0
  - federation: 1.2.0
  - files: 1.7.2
  - files_pdfviewer: 1.1.1
  - files_sharing: 1.4.0
  - files_texteditor: 2.4.1
  - files_trashbin: 1.2.0
  - files_versions: 1.5.0
  - files_videoplayer: 1.1.0
  - firstrunwizard: 2.1
  - gallery: 17.0.0
  - keeweb: 0.4.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.0.0
  - mail: 0.6.4
  - news: 11.0.5
  - nextant: 1.0.8
  - nextcloud_announcements: 1.1
  - notes: 2.2.0
  - notifications: 2.0.0
  - oauth2: 1.0.5
  - passman: 2.1.2
  - password_policy: 1.2.2
  - provisioning_api: 1.2.0
  - richdocuments: 1.12.33
  - serverinfo: 1.2.0
  - sharebymail: 1.2.0
  - spreedme: 0.3.9
  - survey_client: 1.0.0
  - systemtags: 1.2.0
  - tasks: 0.9.5
  - theming: 1.3.0
  - twofactor_backupcodes: 1.1.1
  - updatenotification: 1.2.0
  - weather: 1.4.2
  - workflowengine: 1.2.0
Disabled:
  - encryption
  - files_external
  - user_external
  - user_ldap

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your Nextcloud installation folder
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>
{
    "system": {
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
		***removed system names***
        ],
        "datadirectory": "\/var\/ncdata",
        "overwrite.cli.url": "***removed system name***",
        "dbtype": "mysql",
        "version": "12.0.0.29",
        "dbname": "nextcloud_db",
        "dbhost": "localhost",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "installed": true,
        "instanceid": "ocqvuxzg38az",
        "appstore.experimental.enabled": "true",
        "mail_smtpmode": "smtp",
        "mail_smtpport": "587",
        "mail_smtphost": "***removed system name***",
        "mail_from_address": "nextcloud",
        "mail_domain": "shockley.net",
        "mail_smtpsecure": "tls",
        "preview_libreoffice_path": "\/usr\/bin\/libreoffice",
        "memcache.local": "\\OC\\Memcache\\Redis",
        "filelocking.enabled": true,
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "\/var\/run\/redis\/redis.sock",
            "port": 0,
            "timeout": 0,
            "dbindex": 0,
            "password": "***REMOVED SENSITIVE VALUE***"
        },
        "htaccess.RewriteBase": "\/",
        "mysql.utf8mb4": true,
        "maintenance": false,
        "loglevel": 2
    }
}

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder

or 

Insert your config.php content here. 
Make sure to remove all sensitive content such as passwords. (e.g. database password, passwordsalt, secret, smtp password, …)
```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...
No

**Are you using encryption:** yes/no
No

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
No

### Client configuration
**Browser:**
Any
**Operating system:**
Any

**Comment by nerzhul** [comment]
hi,
those errors are not related with weather app, sorry

**Comment by StevenShockley** [comment]
Thanks for the reply.  They only seem to occur when I reproduce the error, and they're the only messages in the log.  At any rate, I think the error message could be improved, or the app should prompt for an API key if the key is invalid.

**Comment by nerzhul** [comment]
Fixed by 2a693ed1611f5203a27ef71d828b2245e774dab6 (for API key error)
